### PR TITLE
Update ImageShop

### PR DESCRIPTION
**Pull request by davidrglass**
I edited the crop function so that if the the user moves the mouse outside of the bounds of the image it just sets the bounds of the cropbox to the edge of the image. I cleaned up the cropButton code a little. Added some commentary and cleaned up the file. I tested everything and couldn't find any more bugs, but see if you can find anything.

### CODE REVIEW

File: ImageShop [hunk 1]
```diff
@@ -34,20 +34,15 @@ public class ImageShop extends GraphicsProgram {
...
-		}
+		// This condition ensures that the cropbox starts inside the image
+		if (startX < image.getX()) startX = image.getX();
+		if (startX > image.getWidth() + image.getX()) startX = image.getWidth() + image.getX();
+		if (startY < image.getY()) startY = image.getY();
+		if (startY > image.getHeight() + image.getY()) startY = image.getHeight() + image.getY();
+		removeCropbox(); // Removes the old cropBox
+		cropBox = new GRect(startX,startY,0,0);
+		cropBox.setColor(Color.RED);
+		add(cropBox);
```
Comment (givison) [inline ImageShop]: I wanted to be able to remove the cropBox without cropping if you want to

File: ImageShop [hunk 2]
```diff
@@ -58,15 +53,16 @@ public class ImageShop extends GraphicsProgram {
...
+		//Deals with user dragging crop off the image
+		if (x < image.getX()) x = image.getX();
+		if (x > image.getWidth() + image.getX()) x = image.getWidth() + image.getX();
+		if (y < image.getY()) y = image.getY();
+		if (y > image.getHeight() + image.getY()) y = image.getHeight() + image.getY();
+		double width = Math.abs(x - startX);
+		double height = Math.abs(y - startY);
+		x = Math.min(x, startX);
+		y = Math.min(y, startY);
+		cropBox.setBounds(x, y, width, height);
```
Comment (givison) [inline ImageShop]: Super into these edits though

File: ImageShop (hunk 1 above)
Comment (davidrglass) [inline ImageShop]: It already does that if you click anywhere.

Comment (givison) [inline ImageShop]: It places a new 1 pixel cropbox on the edge of the image though. Not a huge problem, but there is still a cropbox..

**Pull request merged**